### PR TITLE
[ticket/13717] Set the assets after rendering the whole template

### PR DESCRIPTION
**Pull request by Nicofuma**
The goal being to be able to call INCLUDECSS/JS from anywhere in any tempalte or event

PHPBB3-13717

### CODE REVIEW

**Comment by Nicofuma** [comment]
@VSEphpbb

File: phpBB/phpbb/template/assets_bag.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\template;
+
+class assets_bag
+{
+	/** @var asset[] */
+	protected $stylesheets = [];
```
Comment (VSEphpbb) [inline phpBB/phpbb/template/assets_bag.php]: Need to be 5.3.3 compatible arrays

File: phpBB/phpbb/template/assets_bag.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	/**
+	 * Returns the HTML code to includes all css assets
+	 *
+	 * @return string
+	 */
+	public function get_stylesheets_content()
+	{
+		$output = '';
+		foreach ($this->stylesheets as $stylesheet) {
```
Comment (VSEphpbb) [inline phpBB/phpbb/template/assets_bag.php]: braces on newline

File: phpBB/phpbb/template/assets_bag.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	/**
+	 * Returns the HTML code to includes all js assets
+	 *
+	 * @return string
+	 */
+	public function get_scripts_content()
+	{
+		$output = '';
+		foreach ($this->scripts as $script) {
```
Comment (VSEphpbb) [inline phpBB/phpbb/template/assets_bag.php]: braces

File: phpBB/phpbb/template/assets_bag.php (hunk 1 above)
Comment (Nicofuma) [inline phpBB/phpbb/template/assets_bag.php]: :cry:

**Comment by nickvergessen** [comment]
Not in 3.1 please, already broke enough with "improvements on the session". Yes, it might be faster or lighter or whatever, but maintenance means less risky.

**Comment by VSEphpbb** [comment]
Not having this in 3.1 will break BC for the many extensions that would take advantage of this. So this has no point IMO if it is not in 3.1 and 3.2.

**Comment by nickvergessen** [comment]
What? We can't break BC, because this simply **never** worked?

I understand your wish to use the same extension in 3.2 as you do in 3.1, then simply you have to use the 3.1 way. BC break would be the other way around (something that worked in 3.1 does not work in 3.2 anymore), but I don't read, that this is the case?

**Comment by Nicofuma** [comment]
replaced by https://github.com/phpbb/phpbb/pull/4131 (for 3.2)

**Comment by lavigor** [comment]
2548 commits here caused that 58 tickets were opened after this PR had been closed.
Only 40 of them were closed after this.

**Comment by Nicofuma** [comment]
do you know which tickets are still open?

**Comment by ErnadoO** [comment]
For exemple: https://tracker.phpbb.com/browse/PHPBB3-14386

**Comment by Nicofuma** [comment]
Found the missing tickets, thanks

**Pull request closed without merging**